runtime/src/loader_utils.rs

```diff
@@ -89,14 +91,14 @@ pub fn load_buffer_account<T: Client>(
         let message = Message::new(
             &[bpf_loader_upgradeable::write(
                 &buffer_pubkey,
-                None,
+                &buffer_authority_pubkey,
                 offset,
                 chunk.to_vec(),
             )],
             Some(&from_keypair.pubkey()),
         );
         bank_client
-            .send_and_confirm_message(&[from_keypair, &buffer_keypair], message)
```

Comment:
Oh did the bank client not fail if you try to sign with an unneeded keypair?

Reply:
Writes require the buffer authority, it was just that the buffer authority was the buffer_keypair.